benchmarks/Bio/BCR.jmd

```diff
@@ -74,78 +81,200 @@ show(to)
 
 ```julia
 sol = solve(oprob, CVODE_BDF(), saveat=tf/1000., reltol=1e-5, abstol=1e-5)
-plot(sol, legend=false, fmt=:png)
+plot(sol; idxs=obs, legend=false, fmt=:png)
 ```
 
-For these benchmarks we will be using the time-series error with these saving
-points since the final time point is not well-indicative of the solution
-behavior (capturing the oscillation is the key!).
-
 ## Generate Test Solution
 
 ```julia
 @time sol = solve(oprob, CVODE_BDF(), abstol=1/10^12, reltol=1/10^12)
-test_sol  = TestSolution(sol)
+test_sol  = TestSolution(sol);
 ```
 
 ## Setups
 
+#### Sets plotting defaults
+
+```julia
+default(legendfontsize=7,framestyle=:box,gridalpha=0.3,gridlinewidth=2.5)
+```
+
+#### Declares a plotting helper function
+
+```julia
+function plot_settings(wp)
+    times = vcat(map(wp -> wp.times, wp.wps)...)
+    errors = vcat(map(wp -> wp.errors, wp.wps)...)
+    xlimit = 10 .^ (floor(log10(minimum(errors))), ceil(log10(maximum(errors))))
+    ylimit = 10 .^ (floor(log10(minimum(times))), ceil(log10(maximum(times))))
+    return xlimit,ylimit
+end
+```
+
+#### Declare pre-conditioners
+```julia
+using IncompleteLU, LinearAlgebra
+
+jaccache = sparsejacprob.f.jac(oprob.u0,oprob.p,0.0)
+W = I - 1.0*jaccache
+prectmp = ilu(W, τ = 50.0)
+preccache = Ref(prectmp)
+
+τ1 = 1e2
+function psetupilu(p, t, u, du, jok, jcurPtr, gamma)
+    if jok
+        sparsejacprob.f.jac(jaccache,u,p,t)
+        jcurPtr[] = true
+
+        # W = I - gamma*J
+        @. W = -gamma*jaccache
+        idxs = diagind(W)
+        @. @view(W[idxs]) = @view(W[idxs]) + 1
+
+        # Build preconditioner on W
+        preccache[] = ilu(W, τ = τ1)
+    end
+end
+function precilu(z,r,p,t,y,fy,gamma,delta,lr)
+    ldiv!(z,preccache[],r)
+end
+
+τ2 = 1e10
+function incompletelu(W,du,u,p,t,newW,Plprev,Prprev,solverdata)
+    if newW === nothing || newW
+        Pl = ilu(convert(AbstractMatrix,W), τ = τ2)
+    else
+        Pl = Plprev
+    end
+    Pl,nothing
+end;
+```
+
+#### Sets tolerances
+
 ```julia
 abstols = 1.0 ./ 10.0 .^ (5:8)
 reltols = 1.0 ./ 10.0 .^ (5:8);
+```
+
+## Work-Precision Diagrams (CVODE and lsoda solvers)
+
+#### Declare solvers.
+
+```julia
+setups = [
+        Dict(:alg=>lsoda(), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(linear_solver=:LapackDense), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(linear_solver=:GMRES), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(linear_solver=:GMRES,prec=precilu,psetup=psetupilu,prec_side=1), :prob_choice => 2),
+        Dict(:alg=>CVODE_BDF(linear_solver=:KLU), :prob_choice => 3),

```

Comment:
So I don't understand why it was added here?

Reply:
I presume that was this failure? https://buildkite.com/julialang/scimlbenchmarks-dot-jl/builds/1142#018473ac-e339-42e6-9a10-5d2cfd70d14f/333-375